server/src/main/java/org/opensearch/action/support/replication/ReplicationProxyFactory.java

```diff
@@ -17,13 +17,15 @@
  */
 public class ReplicationProxyFactory {
 
-    public static <ReplicaRequest> ReplicationProxy<ReplicaRequest> create(
+    public static <ReplicaRequest extends ReplicationRequest<ReplicaRequest>> ReplicationProxy<ReplicaRequest> create(
         final IndexShard indexShard,
-        final ReplicationMode replicationModeOverride
+        final ReplicationMode replicationModeOverride,
+        final ReplicationOperation.Replicas<ReplicaRequest> replicasProxy,
+        final ReplicationOperation.Replicas<ReplicaRequest> primaryTermValidationProxy
     ) {
         if (indexShard.isRemoteTranslogEnabled()) {
-            return new ReplicationModeAwareProxy<>(replicationModeOverride);
+            return new ReplicationModeAwareProxy<>(replicationModeOverride, replicasProxy, primaryTermValidationProxy);
         }
-        return new FanoutReplicationProxy<>();
+        return new FanoutReplicationProxy<>(replicasProxy);
     }
 }
```

Comment:
Can we simplify the hierarchy
At present we have 
```
            ReplicationProxyFactory 
               |                |
ReplicationModeAwareProxy   FanoutReplicationProxy
    |                   |
FanoutReplicationProxy  PrimaryTermValidationProxy
```

Can we change this to

```
            ReplicationProxyFactory 
               |                |
PrimaryTermValidationProxy   FanoutReplicationProxy
```

Reply:
If we remove this from the factory, then we need to have the creation of these proxies outside in TransportReplicationAction. I can make that change. However, that might make the logic concentrated in initialisation of the proxy. Either of these seem fine to me.

Reply:
Made the changes as suggested.